pkg/domain/entities/artifact.go

```diff
@@ -14,6 +14,13 @@ type ArtifactAddOptions struct {
 	ArtifactType string
 }
 
+type ArtifactExtractOptions struct {
```

Comment:
Depending on what we decide about the option semantics, the comments here might need more detail (or to just refer to the man page for details)